mono/metadata/boehm-gc.c

```diff
@@ -1610,24 +1636,31 @@ find_first_unset (guint32 bitmap)
 	return -1;
 }
 
-static void
-handle_data_alloc_entries (HandleData *handles)
+static HandleData*
+handle_data_alloc_entries(int type)
 {
-	handles->size = 32;
+	g_static_assert (sizeof(HandleData) < HANDLE_DATA_ALIGNMENT);
+	g_static_assert (HANDLE_COUNT % CHAR_BIT == 0);
+	HandleData* handles = mono_valloc_aligned (sizeof(HandleData), HANDLE_DATA_ALIGNMENT, MONO_MMAP_READ | MONO_MMAP_WRITE, MONO_MEM_ACCOUNT_SGEN_INTERNAL);
+	handles->type = type;
+	handles->size = HANDLE_COUNT;
 	if (MONO_GC_HANDLE_TYPE_IS_WEAK (handles->type)) {
-		handles->entries = (void **)g_malloc0 (sizeof (*handles->entries) * handles->size);
 		handles->domain_ids = (guint16 *)g_malloc0 (sizeof (*handles->domain_ids) * handles->size);
 	} else {
-		handles->entries = (void **)mono_gc_alloc_fixed (sizeof (*handles->entries) * handles->size, NULL, MONO_ROOT_SOURCE_GC_HANDLE, NULL, "GC Handle Table (Boehm)");
+		mono_gc_register_root((char*) &handles->entries[0], HANDLE_COUNT * sizeof(gpointer), MONO_GC_DESCRIPTOR_NULL, MONO_ROOT_SOURCE_GC_HANDLE, NULL, "GC Handle Data");
 	}
 	handles->bitmap = (guint32 *)g_malloc0 (handles->size / CHAR_BIT);
+
+	return handles;
 }
 
 static gint
 handle_data_next_unset (HandleData *handles)
 {
 	gint slot;
-	for (slot = handles->slot_hint; slot < handles->size / BITMAP_SIZE; ++slot) {
+	gint jon = BITMAP_SIZE;
+	gint what = handles->size / jon;
```

Comment:
Can we remove these two variables? They seem to be left over from debugging.

Reply:
I'll remove these.

Reply:
Done